File: src/apis/cmd.js
import { invoke } from "@tauri-apps/api/core"

export async function cmd_close_app(name) {
    return await invoke('cmd_close_app', { name })
}


export async function cmd_run_app(file) {
    return await invoke('cmd_run_app', { file })
}


export async function cmd_open_url(url) {
    return await invoke('cmd_open_url', { url })
}


export async function cmd_open_folder(path) {
    return await invoke('cmd_open_folder', { path })
}

File: src-tauri/src/apis/cmd.rs
use crate::errors::Result;
use services::cmd;

#[tauri::command(async)]
pub fn cmd_close_app(name: &str) -> Result<()> {
    Ok(cmd::cmd_close_app(name)?)
}

#[tauri::command(async)]
pub fn cmd_run_app(file: &str) -> Result<()> {
    Ok(cmd::cmd_run_app(file)?)
}

#[tauri::command(async)]
pub fn cmd_open_url(url: &str) -> Result<()> {
    Ok(cmd::cmd_open_url(url)?)
}

#[tauri::command(async)]
pub fn cmd_open_folder(path: &str) -> Result<()> {
    Ok(cmd::cmd_open_folder(path)?)
}